### PR TITLE
#add eslint to git actions

### PR DESCRIPTION
**Pull request by BelousSofiya**
Add Eslint to our project, fix frontend code with eslint, add eslint to git actions
Add linter Black to our project, fix backend code with black, add black to git actions

### CODE REVIEW

File: .github/workflows/eslint.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#          node-version: '14'
+#
+#      - name: Code Checkout
+#        uses: actions/checkout@v3
+#
+#      - name: Install Dependencies
+#        run: npm ci
+#
+#      - name: Code Linting
+#        run: npm run lint
```
Comment (ProDG) [inline .github/workflows/eslint.yml:22]: Maybe just remove it, instead of leaving commented config?

File: administration/serializers.py [hunk 2]
```diff
@@ -29,15 +28,15 @@ class Meta(UserSerializer.Meta):
...
+            "is_staff",
         )
         read_only_fields = ("phone_number", "email")
 
 
 class UserSerializer(serializers.ModelSerializer):
     class Meta:
         model = CustomUser
-        fields = ('email', 'name', 'surname')
+        fields = ("email", "name", "surname")
```
Comment (ProDG) [inline administration/serializers.py:39]: I really dislike such approach, quotes reflecting context in a long run make work with code more efficient. But.. as you wish.

Comment (BelousSofiya) [reply administration/serializers.py:39]: It's not me, it's linter

File: authentication/migrations/0001_initial.py [hunk 3]
```diff
@@ -4,31 +4,63 @@
...
-                ('last_login', models.DateTimeField(blank=True, null=True, verbose_name='last login')),
-                ('email', models.EmailField(max_length=50, unique=True)),
-                ('name', models.CharField(max_length=50)),
-                ('surname', models.CharField(max_length=50)),
-                ('is_active', models.BooleanField(default=False)),
-                ('is_staff', models.BooleanField(default=False)),
-                ('is_superuser', models.BooleanField(default=False)),
-                ('groups', models.ManyToManyField(blank=True, help_text='The groups this user belongs to. A user will get all permissions granted to each of their groups.', related_name='user_set', related_query_name='user', to='auth.group', verbose_name='groups')),
-                ('user_permissions', models.ManyToManyField(blank=True, help_text='Specific permissions for this user.', related_name='user_set', related_query_name='user', to='auth.permission', verbose_name='user permissions')),
+                (
```
Comment (ProDG) [inline authentication/migrations/0001_initial.py:17]: I **strongly** recommend to exclude Django-generated files (such as migrations) from Black's scope.

Please fix this before merging PR.

**Review by ProDG — APPROVED** [review]

File: administration/serializers.py (hunk 2 above)
Comment (ProDG) [reply administration/serializers.py:39]: @BelousSofiya I know that it's linter. :)

File: authentication/migrations/0001_initial.py (hunk 3 above)
Comment (BelousSofiya) [reply authentication/migrations/0001_initial.py]: Thank you for feedback. Done)) I fixed it for venv and migrations

**Review by whooaami — APPROVED** [review]

File: FrontEnd/.eslintrc.js
Comment (popovycholeg) [inline FrontEnd/.eslintrc.js:1]: Why do we have two eslint files?

File: FrontEnd/package.json [hunk 4]
```diff
@@ -46,6 +47,13 @@
     ]
   },
   "devDependencies": {
-    "dotenv": "^16.3.1"
+    "dotenv": "^16.3.1",
+    "eslint": "^8.50.0",
+    "eslint-plugin-functional": "^6.0.0",
```
Comment (popovycholeg) [inline FrontEnd/package.json:52]: As far as i see we do not use the plugin

File: FrontEnd/package.json [hunk 5]
```diff
@@ -46,6 +47,13 @@
...
+    "eslint-plugin-import": "^2.28.1",
+    "eslint-plugin-no-null": "^1.0.2",
+    "eslint-plugin-react": "^7.33.2",
+    "eslint-plugin-rxjs": "^5.0.3",
+    "eslint-plugin-storybook": "^0.6.14"
```
Comment (popovycholeg) [inline FrontEnd/package.json:57]: Do we use a storybook at all?

**Review by popovycholeg — APPROVED** [review]

**Pull request merged**